packages/flutter_tools/test/integration.shard/devtools_uri_test.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/convert.dart';
+
+import '../src/common.dart';
+import 'test_data/basic_project.dart';
+import 'test_utils.dart';
+
+void main() {
+  late Directory tempDir;
+  final BasicProject project = BasicProject();
+
+  setUp(() async {
+    tempDir = createResolvedTempDirectorySync('run_test.');
+    await project.setUpIn(tempDir);
+  });
+
+  tearDown(() async {
+    tryToDelete(tempDir);
+  });
+
+  // Regression test for https://github.com/flutter/flutter/issues/126691
+  testWithoutContext('flutter run --start-paused prints DevTools URI', () async {
+    final Completer<void> completer = Completer<void>();
+    final RegExp matcher = RegExp(r'The Flutter DevTools debugger and profiler on');
+
+    final String flutterBin = fileSystem.path.join(getFlutterRoot(), 'bin', 'flutter');
+    final Process process = await processManager.start(<String>[
+      flutterBin,
+      'run',
+      '--start-paused',
+      '-d',
+      'flutter-tester',
+    ], workingDirectory: tempDir.path);
+
+    late StreamSubscription<String> sub;
+    sub = process.stdout.transform(utf8.decoder).listen((String message) {
```

Comment:
Dart is slowly getting too smart for me, but declaring this `late` and then assigning it on the next line seems suspect.

Reply:
I think this can be final when declared on line 42, right?

Reply:
oh wait, maybe it has to be `late final`.

Reply:
It looks like this needs to be late since it's referred to in the closure passed to `listen(...)` (the analyzer complains at line 46 if I remove the late).

I can make this `late final` though, if that's what we want to do.

Reply:
i think `late final` is marginally better than `late`

Reply:
Done.